src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -47,10 +58,16 @@ internal class FindAllReferencesHandler : IRequestHandler<ReferenceParams, VSRef
                 throw new ArgumentNullException(nameof(documentMappingProvider));
             }
 
+            if (lspProgressListener is null)
+            {
+                throw new ArgumentNullException(nameof(lspProgressListener));
+            }
+
             _requestInvoker = requestInvoker;
             _documentManager = documentManager;
             _projectionProvider = projectionProvider;
             _documentMappingProvider = documentMappingProvider;
+            _lspProgressListener = lspProgressListener;
         }
 
         public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
```

Comment:
We're currently ignoring this `cancellationToken`. If the platform cancels the request we probably want to cancel our progress listening. I'd suggest including this in the `TryListenForProgress` signature. Sorry I didn't catch this on the last pass, this will probably be a significant change to the innards of the listener 😢

Reply:
We actually do react to this `cancellationToken`.

Unrelated comment; but I explained the process here how we cancel our progress listening when the platform cancels the request:
https://github.com/dotnet/aspnetcore-tooling/pull/2119/files/0fa823f2b6218c9ea96d02c4028b3b20a521104a#r446381874

Reply:
Ah I overlooked that portion. So my concern with the way you do it currently then is:

1. If the platform cancels the request and Roslyn respects that (because we pass through the cancellation token) that means they're going to stop pushing data; which means every time we'll have to timeout as a "reaction" to the platform cancelling the request. The reason why we'd timeout is because we're currently only checking the cancellation token's cancellation state when we get told about data. If we stop getting told about data we can't "cancel".
2. Every consumer of the progress listener needs to ensure it's passing through the cancellation token that is in its own closure into its own progress report callback method which indirectly couples every implementation with the expectation that the progress listener will "do the right thing"; otherwise every impl will need to timeout.